lib/AST/Expr.cpp

```diff
@@ -2860,6 +2860,13 @@ bool Expr::HasSideEffects(const ASTContext &Ctx,
   case ObjCAvailabilityCheckExprClass:
   case CXXUuidofExprClass:
   case OpaqueValueExprClass:
+  case PositionalParameterExprClass:
```

Comment:
nit: this could be grouped with the 4 bounds expressions after the comment, for clarity.